lhcb-dev.md

```diff
@@ -137,23 +208,88 @@ create a new merge request, for example https://gitlab.cern.ch/lhcb/Stripping/me
+
+It is advisable to test new developments on the so-called [nightly
+builds](https://lhcb-nightlies.cern.ch). Each project is built overnight (hence
+the name), and all pending merge requests are applied. You can use a nightly
+build version of a project with:
+
 ```bash
 lb-dev --nightly lhcb-head DaVinci/HEAD
 ```
+
 A more detailed description of the command is found here:
 
  * [SoftwareEnvTools](https://twiki.cern.ch/twiki/bin/view/LHCb/SoftwareEnvTools)
 
-Sometimes mistakes happen and the committed code is either not compiling or does not do what it is supposed to do.
-Therefore the nightly tests are performed. They first try to build the full software stack.
-If that is successful, they run some reference jobs and compare the output of the new build with a reference file.
-The results of the nightly builds can be found here.
+Sometimes mistakes happen and the committed code is either not compiling or
+does not do what it is supposed to do.  Therefore the nightly tests are
+performed. They first try to build the full software stack.
+
+If that is successful, they run some reference jobs and compare the output of
+the new build with a reference file.  The results of the nightly builds can be
+found here.
 
 * [Nightly builds summaries](https://lhcb-nightlies.cern.ch)
 
-If the aim of the commit was to change the ouput, e.g. because you increased the
-track reconstruction efficiency by a factor of two, mention it in the merge request
-description, such that the manager of the affected project can update the reference file.
+If the aim of the commit was to change the ouput, e.g. because you increased
+the track reconstruction efficiency by a factor of two, mention it in the merge
+request description, such that the manager of the affected project can update
+the reference file.
+
+## Finding stuff
+
+If you want to take a look the source code, without checking it out, you can
+easily access the repository through the [Gitlab web
+interface](https://gitlab.cern.ch/lhcb). This website also provides search
+functionality, but the output is not always easy to read, especially if it
+returns many hits. To search a project much quicker, you can use `Lbglimpse`.
+It allows you to search for a given string in the source code of a particular
+LHCb project.
+
+```bash
+Lbglimpse "PVRefitter" DaVinci v42r2
+```
+This works with every LHCb project and released version. Since it's a shell
+command, you can easily process the output using `less`, `grep`, and other
+tools.
+
+To get an idea of how a certain component of the LHCb software works, you can
+also access the doxygen documentation. One set of doxygen web pages is
+generated for several related projects, and is linked in all the projects web
+sites, like
+[https://cern.ch/LHCb-release-area/DOC/davinci/](https://cern.ch/LHCb-release-area/DOC/davinci/).
+See also the the [LHCb Computing web page](http://cern.ch/lhcb-comp/) for a
+list of projects.
+
+## Working with a full project checkout
+
```

Comment:
Here we should point to https://gitlab.cern.ch/lhcb/upgrade-hackathon-setup  or
to @rmatev development to make working with project checkouts more convenient
https://gitlab.cern.ch/lhcb-HLT/trigger-dev.